File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.java
class Solution {
    public int findMin(int[] nums) {
        int l = 0, r = nums.length-1;
        if (r==0) return nums[0];
        if (nums[l] < nums[r]) return nums[l];

        while (l<=r) {
            int mid = l + (r-l)/2;
            if (nums[mid]>nums[l] || l==mid) {
                if(nums[mid] > nums[mid+1]) return nums[mid+1];
                l = mid+1;
            } else {
                r = mid-1;
            }
        }

        return -1;
    }
}

File: Data Structures & Algorithms/house-robber-ii/submission-1.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
     
        def helper(nums): 
            a, b = 0, 0
            for i in range(len(nums)):
                a, b = b, max(nums[i] + a, b)
            return b
        
        return max(helper(nums[1:]), helper(nums[:-1]))
        